### DISCUSSION TITLE
Difference between "contact" and "profile" avatars

### DISCUSSION THREAD

**Discussion by exquo** [AsamK/signal-cli#874] [Q&A] [answered]
In `~/.local/share/signal-cli/avatars/` some of the contacts' avatar file names start with `contact-` and others with `profile-`. I also have a contact that has both files, identical. What is the difference between them?

In the [source code](https://github.com/AsamK/signal-cli/blob/d812c249baa5ac227a49425375b670d31c8c3584/lib/src/main/java/org/asamk/signal/manager/AvatarStore.java) they seem to be doing the same things, under either "..contact.." or "..profile.." names.

**Comment by AsamK** [comment, marked as answer]
`profile-` avatars are the pictures the other users have set themselves in their Signal profile. (stored encrypted on the Signal servers)
`contact-` avatars are synced from the main device, it used to be the picture from the user's device address book. Now depending on the setting of the main device it can be the same as the user's profile picture.

**Reply by exquo** [reply]
Thanks!
